Chrysalis-Papilio/Soen490-Papilio-Website
# PAPILIO-90-task

## General

The current implementation of the website gives the option to our user to select and delete employee from the dashboard. The behaviour is working as it should however, the employee deletion only happens in the frontend. There are no communication with the backend meaning that once the employee is deleted in the frontend, it is not deleted in the database. This leads to an error because once the dashboard is reloaded, the deleted employee will once again reappear.

_Issue number_: Papilio-90

## Task description
- Create delete employee api call
- Add deleteEmployee to deletion process
- Create associated tests
- Refactor to simplify code

## Acceptance test
### An admin can delete another employee 
GIVEN an authenticated employee
AND the employee is an ADMIN
AND the employee is in the Employee Dashboard
WHEN the employee delete an employee
THEN it does not appear on page reload
 
### An admin can’t delete himself 
GIVEN an authenticated employee
AND the employee is an ADMIN
AND the employee is in the Employee Dashboard
WHEN the employee add himself for deletion
THEN the employee is not added

## Manual Test
To test you need to have an employee account with an Admin role. If you don't have one, you can ask me to provide one.

Make sure that you have the appropriate variables in the backend .env file. (Contact me if you don't)

1. start the backend
`npm run build && npm start`
2. start the website
`npm start`
3. On the main page navigate to the login page
4. Insert the admin account information
5. Once logged in, navigate to the Employee page
6. Click on the delete button
7. On the delete page, try to add your account for deletion. You should not be able to do it.
8. Click on another employee and click on delete button
9. Navigate back to the main page of the Employee Dashboard. The employee you deleted should not be present

##Testing
To run all the unit tests, run the following code: `npm test`
To make sure that the tests have reach the appropriate level of coverage, run the following code: `npm run coverage`
